addons/account/models/account_analytic_line.py

```diff
@@ -87,3 +87,42 @@ def view_header_get(self, view_id, view_type):
                 account=self.env['account.analytic.account'].browse(self.env.context['account_id']).name
             )
         return super().view_header_get(view_id, view_type)
+    
+    def _compute_key(self, analytic_line):
+        key = []
+        if analytic_line.account_id:
+            key.append(str(analytic_line.account_id.id))
+        key.extend(
+            str(analytic_line[field_name].id)
+            for field_name in analytic_line._get_plan_fnames()
+            if field_name.startswith("x_plan") and analytic_line[field_name]
+        )
+        return ",".join(key)
```

Comment:
You don't need to create the list before, you can just pass the generator as argument to `str.join`